datamodel/samples.d/generate_timestampdefinitions.py

```diff
@@ -0,0 +1,1188 @@
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=EST_REQ_PLN,
+    )
+
+    #Towage UC 58 - 60 + UC 64 - 66
+    xty_service_timestamps(
+        EST_REQ_PLN,
+        ['TOWG'],
+        ['OUTB', 'SHIF'],
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=EST_REQ_PLN,
+    )
+
+    #ATC Bunkering UC 69
+    generate_special_timestamp(
+        'ATC Bunkering',
+        as_publisher_patterns(['BUK'], ['CA']),
+        'CMPL',
+        'BRTH',
+        'ALGS',
+        "Port Departure Planning And Services Completion",
+        'jit1_1',
+        'BUNK',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
+
+    #ATC Cargo Ops Load UC 70
+    generate_special_timestamp(
+        'ATC cargo ops Load',
+        PUBLISHER_PATTERN_TR2CA,
+        'CMPL',
+        'BRTH',
+        'ALGS',
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        'LCRO',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
+
+    #ATC Cargo Ops UC 71
+    generate_special_timestamp(
+        'ATC cargo ops',
+        PUBLISHER_PATTERN_TR2CA,
+        'CMPL',
+        'BRTH',
+        'ALGS',
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        'CRGO',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
+
+    #Mooring UC 72 - 77
+    xty_service_timestamps(
+        EST_REQ_PLN,
+        ['MOOR'],
+        ['OUTB'],
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        include_phase_in_name=True,
+        is_cancelable=True,
+        need_event_location_for=EST_REQ_PLN,
+    )
+
+    #ATC Lashing UC 78
+    generate_special_timestamp(
+        'ATC Lashing',
+        as_publisher_patterns(['MOR'], ['CA']),
+        'CMPL',
+        'BRTH',
+        'ALGS',
+        "Port Departure Planning And Services Completion",
+        'jit1_2',
+        'LASH',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
+
+    #Terminal Ready for vessel departure UC 79
+    generate_special_timestamp(
+        'Terminal Ready for vessel departure',
+        PUBLISHER_PATTERN_TR2CA,
+        'DEPA',
+        'BRTH',
+        'ALGS',
+        "Port Departure Execution",
+        'jit1_1',
+        'SAFE',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
+
+    #Vessel ready to sail UC 80
+    generate_special_timestamp(
+        'Vessel Ready to sail',
+        PUBLISHER_PATTERN_CA2TR,
+        'DEPA',
+        'BRTH',
+        'ALGS',
+        "Port Departure Execution",
+        'jit1_1',
+        'VRDY',
+        need_vessel_position=False,
+        need_event_location=True,
+    )
```

Comment:
This variant is JIT 1.2.

There is also a JIT 1.1 variant that uses `SAFE` instead of `VRDY` (so we need this block twice).

Reply:
added the SAFE variant